Ws 64 gamification toggle

turrilld opened the pull request:
Added: Gamification button in editor now swaps and saves the gamification state, either true or false.  If true, the gamification elements are shown, if false, they are hidden.  Weather or not the VFE is gamified is now exported to the save file and loaded when the file is loaded.  

Known Bugs:  
- Loading the VFE into the viewer works as expected.  Loading the VFE into the editor will use whatever gamification state is currently in local memory to render the page and only change to the correct state if the page is reloaded.  Uncertain how this is possible at this time.  Timing issue?  Not setting the state early enough? 
- Minor bug setting the gamification state for saving....it was always saving the opposite state it should have.  Currently corrected for in code, but I am seeking a better solution than simply setting the saved state to what it should be.

Carl-Swin reviewed (approved):
Love the additions. I especially like how it can be exported with it enabled for the viewer. For the gamification state being set to whatever is currently in local memory, using something like sessionStorage might fix it.

Braulee7 reviewed (approved):
Changes are good. Only thing I may add is a "reset points button" or automatically reset them when you disable and re-enable the feature to allow users to debug and play around better. Looks promising so far though !

turrilld commented on src/Pages/PageUtility/PointsInterface.tsx:
```diff
@@ -58,3 +74,48 @@ export function usePoints() {
+    } else {
+      console.log("No State, intialize!");
+      InitializeState();
+      return;
```
Edited and updated to ensure a state is always set and the button should only need to be pressed once even when initializing a new state.

turrilld commented:
Added a reset points button, cleaned up the points logic to prevent the need to hit the button several times on initialization.  Will merge tomorrow if I still cant figure out whats causing the bug with loading to the editor.
